### PR TITLE
Feature/subscribe channel new visit notification

### PR DESCRIPTION
**Pull request by yashuk803**

### CODE REVIEW

File: frontend/src/store/modules/dashboard/actions.js [hunk 1]
```diff
@@ -49,63 +52,28 @@ export default {
...
-       });
+        context.commit(SET_BUTTON_FETCHING);
 
-       const result = [];
-        items.forEach((element) => {
-            if(result.length > 0) {
-                if(!result.find( (item => item.url === element.url && item.visitorId === element.visitorId))) {
+        return getActivityDataItems().then(response => {
+            response.sort( (a, b) => {
+                return  a.date - b.date || a.url - b.url || a.visitor - b.visitor;
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/dashboard/actions.js]: are you sure you can subtract urls?

File: frontend/src/store/modules/dashboard/actions.js [hunk 2]
```diff
@@ -49,63 +52,28 @@ export default {
...
+            });
+
+            const result = [];
+            response.forEach((element) => {
+                if(result.length > 0) {
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/dashboard/actions.js]: in empty array you also won't find anything, so I think this condition doesn't have any sense

File: frontend/src/components/dashboard/home/ActiveVisitorsCard.vue [hunk 3]
```diff
@@ -82,12 +82,22 @@
             gradientDirection: 'left',
             polling: null,
         }),
+        mounted() {
+            const channel = echoInstance.private('private-active-users.'+ this.website.id);
+            channel.listen('VisitCreated', function(data) {
+                console.log(data);
```
Comment (lenchvolodymyr) [inline frontend/src/components/dashboard/home/ActiveVisitorsCard.vue]: I am not sure you need console.log)

File: frontend/src/store/modules/dashboard/actions.js [hunk 4]
```diff
@@ -114,9 +80,18 @@ export default {
...
+            moment().diff(moment(item.date), 'minutes') < 5
         );
         context.commit(SET_ACTIVITY_DATA_ITEMS, data);
     },
 
+    [REFRESH_ACTIVITY_DATA_ITEMS]: (context, data) => {
+        const items =  context.state.activityData.items;
+        const index = _.indexOf(items, _.find(items, {url: data.url, visitor: data.visitor}));
+        if(index >= 0) {
+            items.splice(index, 1);
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/dashboard/actions.js]: I am not sure you can mutate state inside the action. You should do it inside the mutator I suppose. @TatsuUkraine am I right?

Comment (TatsuUkraine) [reply frontend/src/store/modules/dashboard/actions.js]: Yep

Comment (TatsuUkraine) [reply frontend/src/store/modules/dashboard/actions.js]: I would also replace index searching approach with findIndex method

Comment (TatsuUkraine) [reply frontend/src/store/modules/dashboard/actions.js]: or event replace it with filter and spread operator

**Pull request merged**